### PR TITLE
Experiment with streamlined Jenkins builds

### PR DESCRIPTION
**Pull request by rbradford**
- build: Run linter over Jenkinsfile
- build: Consolidate integration testing to reused built assets

### CODE REVIEW

**Comment by rbradford** [comment]
Looks like we need to handle a clippy update.

**Comment by likebreath** [comment]
> Looks like we need to handle a clippy update.

@rbradford I will submit a PR to take care of that today.

File: scripts/run_integration_tests_x86_64.sh [hunk 1]
```diff
@@ -124,7 +124,9 @@ popd
 
 # Build custom kernel based on virtio-pmem and virtio-fs upstream patches
 VMLINUX_IMAGE="$WORKLOADS_DIR/vmlinux"
-build_custom_linux
+if [ ! -f "$VMLINUX_IMAGE" ]; then
+    build_custom_linux
+fi
```
[the remark this answers is not in the document]
Comment (rbradford) [reply scripts/run_integration_tests_x86_64.sh:129]: No, because this is the x86_64 integration test script.

**Comment by sboeuf** [comment]
@rbradford looking good, and it only bumped the CI time by 10min as we expected. Given we're saving the cost of 2 extra machines for every PR, I'd say this is worth it!

**Review by sboeuf — APPROVED** [review]

**Pull request merged**